### PR TITLE
Automations Project

### LINKED ISSUE #1228: [AUTOMATION] Add audited version of scheduler. And add first version of autowrap project.
Todo:

- [x] Add deployment scripts with instructions
- [x] Deploy to testnets / mainnet, including verifying the contracts.
- [x] Update README of both projects
- [ ] Add audit result PDF file, and upload to etherscans.
- [x] Delete https://github.com/superfluid-finance/platform-monorepo/tree/master/packages/evm-contracts/scheduler
- [x] Update metadata repo

Stretch goals:

- [x] Run GitActions tests

Ref:

https://github.com/superfluid-finance/platform-monorepo/pull/85
https://github.com/superfluid-finance/metadata/pull/22

### PR DESCRIPTION
**Pull request by ngmachado**
This PR includes new smart contract for automations.

At this time we are dealing with two subprojects:

- autowrap
- scheduler (vesting and flow)

This will include a new package called `automation-contracts`.
Smart contracts follow monorepo lint definitions.
Tests are in Foundry.

note: Changed repo global `package.json` for adding new package to workspace.

### CODE REVIEW

**Comment by vmichalik** [comment]
Closes https://github.com/superfluid-finance/platform-monorepo/issues/33

File: .github/workflows/ci.feature.yml [hunk 1]
```diff
@@ -108,6 +110,90 @@ jobs:
...
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Test automation-contracts-scheduler
+        run: |
+          yarn workspace scheduler test
+
+  test-automation-contracts-autowrap:
+    name: Build and test automation-contracts-autpwrap (Feature Branch)
```
Comment (0xdavinchee) [inline .github/workflows/ci.feature.yml:157]: minor typo, should be autowrap

File: .github/workflows/ci.feature.yml [hunk 2]
```diff
@@ -105,6 +107,90 @@ jobs:
...
```
Comment (0xdavinchee) [inline .github/workflows/ci.feature.yml]: small typo

File: .github/workflows/ci.feature.yml [hunk 3]
```diff
@@ -105,6 +107,90 @@ jobs:
...
+
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Test automation-contracts-autowrap
+        run: |
+          yarn workspace autowrap test
+
```
Comment (0xdavinchee) [inline .github/workflows/ci.feature.yml:116]: I forgot to mention that we need to add: `test-automation-contracts-scheduler` and `test-automation-contracts-autowrap` to the `needs` array in the `all-packages-tested` job.

File: .github/workflows/ci.feature.yml [hunk 4]
```diff
@@ -105,6 +107,90 @@ jobs:
...
+    strategy:
+      matrix:
+        node-version: [ 16.x ]
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Check changeset
```
Comment (0xdavinchee) [inline .github/workflows/ci.feature.yml]: please delete this step, it is duplicated and unused in the job, I forgot to remove this from the ethereum-contracts job.

File: .github/workflows/ci.feature.yml [hunk 5]
```diff
@@ -105,6 +107,90 @@ jobs:
...
```
Comment (0xdavinchee) [inline .github/workflows/ci.feature.yml]: please delete this step, it is duplicated and unused in the job, I forgot to remove this from the ethereum-contracts job.

**Review by 0xdavinchee — COMMENTED** [review]
Looks great overall, just a few small things to address. Don't worry, no naming comments 😉

**Comment by hellwolf** [comment]
should we just add those to ci.canary too? seems a low hanging fruit

**Review by 0xdavinchee — APPROVED** [review]
LFG 🔥

**Review by hellwolf — APPROVED** [review]

**Pull request merged**